corehq/apps/registration/utils.py

```diff
@@ -195,7 +195,7 @@ def request_new_domain(request, form, org, domain_type=None, new_user=True):
         dom_req.activation_guid = uuid.uuid1().hex
 
     name = form.cleaned_data['hr_name']
-    with CriticalSection(['request_domain_name_{}'.format(name)]):
+    with CriticalSection([u'request_domain_name_{}'.format(name)]):
```

Comment:
Good point. Addressed this by calling `name_to_url` so that names are stripped of unicode before they're used as keys. It's a bit redundant, because `name_to_url` will get called again inside `generate_name`, but it's safer - as the code was, "हिन्दी" and "हिन्दीहिन्दी" would be different keys, but both would ultimately be turned into "project" and could cause a conflict.